recipe/meta.yaml

```diff
@@ -22,10 +22,9 @@ requirements:
     - make  # [not win]
   host:
     - eigen
-    - boost-cpp
+    - libboost-devel
```

Comment:
It seems that eiquadprog actually only needs the boost headers, as indicated by the link check:
```
WARNING (eiquadprog): dso library package conda-forge::libboost-1.82.0-h6fcfa73_5 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
```

So we could (and very likely should) simplify this to `libboost-headers`, to avoid unnecessarily pulling in all the boost libraries when they're actually not needed.